src/os/bluestore/BlockDevice.h

```diff
@@ -47,6 +47,7 @@ struct IOContext {
 public:
   CephContext* cct;
   void *priv;
+  size_t shard_hint = 0;
```

Comment:
The `0` shard is busier than others: it handles all writes and some of async read completions. Separating those responsibilities is a way to go. At the price of one extra thread we might:

* get better throughput in mixed scenarios,
* be less error-prone -  currently directing any write to any non-`0` shared effects in  `flush` issue,
* introduce a variant of `_aio_thread` optimized for read-only use case.